packages/emitter-framework/src/typescript/components/union-expression.tsx

```diff
@@ -1,36 +1,39 @@
-import { Children, mapJoin } from "@alloy-js/core";
+import * as ay from "@alloy-js/core";
+import { Children } from "@alloy-js/core";
 import * as ts from "@alloy-js/typescript";
-import { Enum, Union } from "@typespec/compiler";
-import { TypeExpression } from "./type-expression.js";
+import { Enum, EnumMember, Union, UnionVariant } from "@typespec/compiler";
+import { $ } from "@typespec/compiler/experimental/typekit";
+import { TypeExpression } from "./type-expression.jsx";
 
 export interface UnionExpressionProps {
   type: Union | Enum;
   children?: Children;
 }
 
 export function UnionExpression({ type, children }: UnionExpressionProps) {
-  let variants: any[];
+  const items = ($.union.is(type) ? type.variants : type.members) as Map<
+    string,
+    UnionVariant | EnumMember
+  >;
 
-  if (type.kind === "Enum") {
-    variants = mapJoin(
-      type.members,
-      (_, value) => {
-        return <ts.ValueExpression jsValue={value.value ?? value.name} />;
-      },
-      { joiner: " | " },
-    );
-  } else {
-    variants = mapJoin(
-      type.variants,
-      (_, variant) => {
-        return <TypeExpression type={variant.type} />;
-      },
-      { joiner: " | " },
-    );
-  }
+  const variants = (
+    <ay.For joiner={" | "} each={items}>
```

Comment:
Formatting here can be improved, but Alloy should probably get a `UnionTypeExpression` which handles that.